components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1255,32 +1226,47 @@ void KeyringService::RemoveSelectedAccountForCoin(
   NotifySelectedAccountChanged(coin);
 }
 
-void KeyringService::RemoveImportedAccount(
-    mojom::CoinType coin,
-    const std::string& keyring_id,
-    const std::string& address,
-    const std::string& password,
-    RemoveImportedAccountCallback callback) {
-  if (address.empty() || !ValidatePasswordInternal(password)) {
+void KeyringService::RemoveAccount(mojom::AccountIdPtr account_id,
+                                   const std::string& password,
+                                   RemoveAccountCallback callback) {
+  if (account_id->kind == mojom::AccountKind::kImported &&
+      !ValidatePasswordInternal(password)) {
```

Comment:
Removal of hw accounts did not validate password at backend side, but password validation is forced by ui